backend/hitas/calculations/max_price.py

```diff
@@ -0,0 +1,318 @@
+            "completion_date",
+            "debt_free_purchase_price",
+            "primary_loan_amount",
+            "apartment_type__value",
+            "building__real_estate__housing_company__id",
+            "building__real_estate__housing_company__official_name",
+            "building__real_estate__housing_company__postal_code__value",
+            "building__real_estate__housing_company__postal_code__city",
+            "building__real_estate__housing_company__property_manager__name",
+            "building__real_estate__housing_company__property_manager__street_address",
+        )
+        .select_related(
+            "apartment_type",
+            "building__real_estate__housing_company",
+            "building__real_estate__housing_company__property_manager",
+            "building__real_estate__housing_company__postal_code",
+        )
+        .prefetch_related(
+            Prefetch(
+                "ownerships",
+                Ownership.objects.only("apartment_id", "percentage", "owner__name").select_related("owner"),
+            ),
+            Prefetch(
+                "construction_price_improvements",
+                ApartmentConstructionPriceImprovement.objects.only("value", "apartment_id", "depreciation_percentage"),
+            ),
+            Prefetch(
+                "market_price_improvements",
+                ApartmentMarketPriceImprovement.objects.only("value", "apartment_id"),
+            ),
+            Prefetch(
+                "building__real_estate__housing_company__construction_price_improvements",
+                HousingCompanyConstructionPriceImprovement.objects.only("value", "housing_company_id",).extra(
+                    select={
+                        "completion_date_index": """
+    SELECT completion_date_index.value
+    FROM hitas_constructionpriceindex2005equal100 AS completion_date_index
+    WHERE completion_date_index.month
+         = DATE_TRUNC('month', hitas_housingcompanyconstructionpriceimprovement.completion_date)
+    """,
+                    },
+                ),
+            ),
+            Prefetch(
+                "building__real_estate__housing_company__market_price_improvements",
+                HousingCompanyMarketPriceImprovement.objects.only("value", "housing_company_id",).extra(
+                    select={
+                        "completion_date_index": """
+    SELECT completion_date_index.value
+    FROM hitas_marketpriceindex2005equal100 AS completion_date_index
+    WHERE completion_date_index.month
+         = DATE_TRUNC('month', hitas_housingcompanymarketpriceimprovement.completion_date)
+    """,
+                    },
+                ),
+            ),
+        )
+        .extra(
+            select={
+                "calculation_date_cpi_2005eq100": """
+SELECT calculation_date_index.value
+FROM hitas_constructionpriceindex2005equal100 AS calculation_date_index
+WHERE calculation_date_index.month = DATE_TRUNC('month', %s)
```

Comment:
Since this is getting used quite a bit, it might be useful to make this a django ORM filter out of this, to get rid of some raw SQL queries in favour of Django ORM queries